lib/r10k/puppetfile.rb

```diff
@@ -212,15 +221,22 @@ def concurrent_accept(visitor, pool_size)
   def modules_queue(visitor)
     Queue.new.tap do |queue|
       visitor.visit(:puppetfile, self) do
-        modules.each { |mod| queue << mod }
+        modules_by_cachedir = modules_by_vcs_cachedir.clone
```

Comment:
Am I missing something here or should `modules_by_vcs_cachedir` be an instance variable or is this appropriate since it's available as an `attr_reader`?

Reply:
Correct, this calls the `attr_reader` method for which the default implementation is to simply return the instance variable of the same name. We added the reader so we could retrieve that value in the tests. This could just as easily be `@modules_by_vcs_cachedir.clone`, since we're in the class where the variable is definied, but as I understand it, it's generally good practice to use the reader method if one is provided. Makes it easier to override with a custom getter later if you end up needing to do something special when retrieving the value.